lib/ansible/errors/__init__.py

```diff
@@ -220,7 +220,24 @@ class AnsibleUndefinedVariable(AnsibleRuntimeError):
 
 class AnsibleFileNotFound(AnsibleRuntimeError):
     ''' a file missing failure '''
-    pass
+
+    def __init__(self, message="", obj=None, show_content=True, suppress_extended_error=False, orig_exc=None, paths=None, file_name=None):
+
+        self.file_name = file_name
+        self.paths = paths
+
+        if self.file_name:
+            if message:
+                message += "\n"
+            message += "Could not find or access '%s'" % to_text(self.file_name)
+
+        if self.paths and isinstance(self.paths, list):
+            searched = to_text('\n\t'.join(self.paths))
+            if message:
+                message += "\n"
+            message += "Searched in:\n\t%s" % searched
```

Comment:
If I were writing the exceptions from scratch, I think I would tend to defer most of the message formatting to the __repr__/__str__ if possible. But I'm not sure how intertwined the the message creation in the base AnsibleError __init__ is required, so there may be reasons we can't defer to __repr__...   ie, if there is code explicitly using AnsibleError.message instead of the str/repr results 

It looks like there are some references to some_exception.message, though at first inspection I dont see any that would do that for an AnsibleError, so moving that to the repr/str might work.